core/sagas/inputs.js

```diff
@@ -73,23 +75,68 @@ function flattenInput(response) {
       release: item.builds[0].release,
       arch: [item.builds[0].arch]
     };
-    if (previousBuilds.hasOwnProperty(build.version + build.release)) {
-      // if this item has the same values for release and version of a
-      // previousBuild
-      // then push arch to the array of arch's, and filter this item out
-      previousBuild = previousBuilds[build.version + build.release];
+    if (previousBuilds.hasOwnProperty(build.version)) {
+      // if this item has the same version value as a
+      // previousBuild replace the release value (this is assumed to be the latest)
+      // then push arch to the array of arch's and filter this item out
+      previousBuild = previousBuilds[build.version];
       previousBuild.arch = previousBuild.arch.concat(build.arch);
+      previousBuild.release = build.release;
       return false;
     } else {
       // else push the build to the array of builds
       flattened.builds = [build].concat(flattened.builds);
     }
-    previousBuilds[build.version + build.release] = build;
+    previousBuilds[build.version] = build;
     return true;
   });
+  flattened.builds = addWildcardVersions(flattened.builds);
   return flattened;
 }
 
+function addWildcardVersions(builds) {
+  // create a new array of builds that includes wildcard versions
+  let wildcardBuilds = [];
+  builds.forEach(item => {
+    item.depsolveVersion = item.version;
+    let newBuilds = [item];
+    const wildcardVersion = createWildcardVersion(item.version);
+    // add this version if it doesn't exist already in wildcardBuilds
+    const versionIndex = wildcardBuilds.findIndex(build => build.version === wildcardVersion);
+    if (versionIndex === -1) {
+      const wildcardBuild = Object.assign({}, item, {
+        version: wildcardVersion
+      });
+      newBuilds.unshift(wildcardBuild);
+    }
+    // if wildcardVersion has a period, repeat this step for each period
+    const versionValues = wildcardVersion.split(".");
+    versionValues.pop();
+    for (let index = 0; index < versionValues.length; index++) {
+      const newWildcardVersion = createWildcardVersion(newBuilds[0].version);
+      const versionIndex = wildcardBuilds.findIndex(build => build.version === newWildcardVersion);
+      if (versionIndex === -1) {
+        const newWildcardBuild = Object.assign({}, item, {
+          version: newWildcardVersion
+        });
+        newBuilds.unshift(newWildcardBuild);
+      }
+    }
+    wildcardBuilds = wildcardBuilds.concat(newBuilds);
```

Comment:
This is a bit complex and inefficient, because it's iterating over the list of versions many times. If I understand what this is doing correctly, I think it can be simplified to something like this:
```javascript
function addWildcardVersions(builds) {
  // create a new array of builds that includes wildcard versions
  let wildcardBuilds = {}; 
  builds.forEach(item => {
    item.depsolveVersion = item.version;

    let parts = item.version.split(".");
    for (let i = 0; i < parts.length; i += 1) {
      let wildcard = parts
        .slice(0, i)
        .concat("*")
        .join(".");

      if (!(wildcard in wildcardBuilds)) {
        wildcardBuilds[wildcard = Object.assign({}, item, { version: wildcard }); 
      }   
    }   
  }); 

  return Object.values(wildcardBuilds);
}
```